Add Create Graph functionality

m-skaan opened the pull request:
1. Add Create Graph button and webpage to home screen
2. Visualize Default Graph on page load.
3. Take user-inputted Adjacency List in given format and update default graph edges accordingly.
4. Show updated graph with Logical Representation, AdjList Representation, and AdjMatrix representation(can toggle with radiolist buttons).

This PR adds creates the basic functionality for users wishing to create a graph. Possible further extensions include:
- allowing for AdjMatrix user input.
- animating edge generation.
- creating a downloadable format for use in the other graph algorithm webpages.
- extending functionality to allow users to draw edges from other node placements for the small graph.
- extending functionality to make more than 8 node graphs.
- hiding nodes with no edge connections.
- further error-checking of user input.
(and more that I can't think of right now.)

There are a lot more changes listed because of the prettier command. My second-to-last commit is more indicative of the changes I made.

luciankt commented:
Amazing work!!

I like having a separate page from the home screen where you can just build graphs. It's good practice for learning how adjacency lists work.

I think the most important thing we can do for usefulness is to have it integrate with the other graph pages, so perhaps there's a "Create Graph" button on each of the graph pages which opens a big Modal of basically this page, where the user can build the graph in the same way, and then click "Done" and it puts that graph data into the page they're currently on. Without having looked at the code very closely, this doesn't seem _too_ hard if you're able to put the Adjacency List input and graph visualization into a React Component, then put that component as part of the AlgoScreen screen, and have a button to open it up only on the graph algo pages (in the algo folder). In terms of having a downloadable format, I think it would be a bit arduous to have the user download and reupload graph data, so I much prefer this modal idea. But we could have users download PNG images of the graph, that could be useful, if you want to do something with downloads!

One more big thing that I think would be amazing is to have directed graphs too. That's an option on all the other graph pages currently so it would be awesome if this can produce those as well!

Comments on your other ideas:
- AdjMatrix: I think this would be nice but a low priority since we don't focus on it in class
- Animating edges: Same thing, would be a great touch but might be a bit tricky
- Downloading format: see above
- Click to draw: I'd love this!!! Probably pretty difficult though.
- More nodes: This would be great too. I don't know how hard it is with the current Graph class implementation but if possible it would be cool to allow any number of nodes? (Maybe cap it at something reasonable like 20-25 though)
- Hiding nodes with no edges: I don't think we should do this actually. Graphs can be disconnected
- Error checking: Always a good thing to do. I think it's pretty intuitive as-is though

m-skaan commented:
Added starter code for modals. Currently commented out. The commented code currently has issues transferring the graph created in the modal to the base webpage once alterations are done.

Fixed a bug where all nodes A-H would appear despite allowance for use of subset in graph generation. e.g: a graph using nodes A-E would still word have had F-H nodes appear as disconnected even though they weren't used.
